web/src/features/charts/graphUtils.ts

```diff
@@ -71,26 +71,38 @@ export const getGenerationTypeKey = (name: string): GenerationType | undefined =
   return undefined;
 };
 
-export function getTotalElectricity(
-  zoneData: ZoneDetail,
-  displayByEmissions: boolean,
-  mixMode: Mode
-) {
-  const includeImports = mixMode === Mode.CONSUMPTION;
-  let productionValue: number;
-
-  if (displayByEmissions) {
-    productionValue = zoneData.totalCo2Production + zoneData.totalCo2Discharge;
-    productionValue += includeImports ? zoneData.totalCo2Import : 0;
-  } else {
-    productionValue = zoneData.totalProduction + zoneData.totalDischarge;
-    productionValue += includeImports ? zoneData.totalImport : 0;
+export function getTotalElectricity(zoneData: ZoneDetail, mixMode: Mode) {
+  const isConsumption = mixMode === Mode.CONSUMPTION;
+  if (isConsumption) {
+    if (zoneData.totalConsumption === null) {
+      return Number.NaN;
+    }
+    return zoneData.totalConsumption;
   }
-  if (productionValue === null) {
+  if (zoneData.totalProduction === null) {
     return Number.NaN;
   }
+  // Electricity: Handle discharge case if available, else default to production
+  return zoneData.totalDischarge !== null && zoneData.totalDischarge !== undefined
+    ? zoneData.totalProduction + zoneData.totalDischarge
+    : zoneData.totalProduction;
+}
```

Comment:
I think this can't be put more simply like this :)

```suggestion
return zoneData.totalProduction + (zoneData.totalDischarge II 0)
}
```

Reply:
And same with emissions function below